core/src/main/java/org/apache/iceberg/BaseTableScan.java

```diff
@@ -135,11 +143,14 @@ public CloseableIterable<FileScanTask> planFiles() {
             planningDuration.stop();
             ScanReport scanReport =
                 ImmutableScanReport.builder()
-                    .projection(schema())
+                    .schemaId(schema().schemaId())
+                    .projectedFieldIds(projectedFieldIds)
+                    .projectedFieldNames(projectedFieldNames)
                     .tableName(table().name())
                     .snapshotId(snapshot.snapshotId())
                     .filter(ExpressionUtil.sanitize(filter()))
                     .scanMetrics(ScanMetricsResult.fromScanMetrics(scanMetrics()))
+                    .metadata(context().options())
```

Comment:
when `TableScanContext#options()` are being used, then it's mainly to configure/pass `TableProperties.SPLIT_OPEN_FILE_COST` / `TableProperties.SPLIT_LOOKBACK` / `TableProperties.SPLIT_SIZE`, so I think it's ok to just add whatever was passed to `TableScanContext#options()` as the metadata for the scan report (since it might be helpful information), wdyt @rdblue?

Reply:
I think it's good to send those, but I wouldn't mix it into this PR to avoid delays.

Why call this `metadata`? That seems to generic. Why not `options` since that is what the scan is using?

Reply:
the idea was here to use a fairly generic term since we could also send more things in the future than just the options that were passed to a scan

Reply:
Ah, okay. So this is sending back additional properties that may include other things. Are we setting the Spark ID in options?

Reply:
yep correct and yep we're sending the Spark ID + execution ID (if available) through the options as that seemed to be the easiest approach to pass things along to the Scan itself

Reply:
I don't think that we should add app ID to options. It causes this `metadata` naming problem and isn't a correct use of options in the first place. Is there a better way that we can pass app ID? Maybe we should make that an optional part of the scan report.

Reply:
I think passing the Spark App ID through the `TableScanContext` is probably the easiest way without changing any APIs. I'm aware that this isn't a strict use-case for `options` but at the same time the `options` seem flexible enough to allow such a use case. 
Optionally, we could maybe add a `metadata()` method to `TableScanContext` to keep it separate from `options()`?

Reply:
as discussed offline, I introduced an `EnvironmentContext` class that allows setting global and thread local properties. That way we can set & retrieve things like the spark app id or the SQL execution id